src/box/decimal.h

```diff
@@ -0,0 +1,574 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include "trivia/util.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/** \cond public */
+
+/* {{{ decimal structure and constants */
+
+/** See @ref box_decimal_t. */
+struct box_decimal {
+	/**
+	 * Decimal data.
+	 *
+	 * The format is unspecified and can be changed.
+	 *
+	 * The tail of the array may be in an unallocated memory.
+	 * Don't access this array directly. Use box_decimal_*()
+	 * functions instead.
+	 */
+	uint64_t internal[8];
```

Comment:
Will someone ask to extend it to `uint64_t[16]` (128 bytes) before we'll proceed with the pull request? With 64 bytes we can't store `uint512_t` range. @Mons?

*(Personally I don't like so fat padding — 64 bytes is already large, but I can't propose anything more viable.)*

Reply:
It seems, no. Okay, let's stand on 64 bytes.